src/appleseed.shaders/src/appleseed/as_cdl.osl

```diff
@@ -0,0 +1,109 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// American Society of Cinematographers Color Decision List implementation
+
+shader as_cdl
+[[
+    string help = "Slope/Offset/Power Color Decision List modifier according to ASC",
+    string icon = "asCDL.png",
+    string URL = "please fill in",
+    
+    string as_node_name = "asCDL",
+    string as_category = "color",
+    
+    string as_max_class_id = "1669361528 687027596", 
+    string as_max_plugin_type = "texture",
+    
+    int as_maya_type_id = 0x00127a07,
+    string as_maya_classification = "drawdb/shader:rendernode/appleseed/utility"
+]]
+(
+    color in_color = color(0)
+    [[
+        string as_maya_attribute_name = "color",
+        string as_maya_attribute_short_name = "c",
+        string label = "Input Color",
+        string page = "Color",
+        string help = "Input color.",
+        int divider = 1
+    ]],
+    float in_slope = 1.0
+    [[
+        string as_maya_attribute_name = "slope",
+        string as_maya_attribute_short_name = "slo",
+        float min = 0.0,
+        float max = 10000.0,
+        string label = "Slope",
+        string page = "Correction",
+        string help = "Slope (Adjusts the highlights).",
+    ]],
+    float in_offset = 0.0
+    [[
+        string as_maya_attribute_name = "offset",
+        string as_maya_attribute_short_name = "ofs",
+        float min = -1.0,
+        float max = 1.0,
+        string label = "Offset",
+        string page = "Correction",
+        string help = "Constant offset (Adjusts the overall brightness).",
+    ]],
+    float in_power = 1.0
+    [[
+        string as_maya_attribute_name = "power",
+        string as_maya_attribute_short_name = "pow",
+        float min = 0.0,
+        float max = 10000.0,
+        string label = "Power",
+        string page = "Correction",
+        string help = "Over-all exponent (Adjusts the midtones).",
+    ]],
+    float in_saturation = 1.0
+    [[
+        string as_maya_attribute_name = "saturation",
+        string as_maya_attribute_short_name = "sat",
+        float min = 0.0,
```

Comment:
Can you also add the *softmin* and *softmax* parameter respectively? This will restrict the range manipulation via sliders in the GUI (in the Maya case), but the user can still enter numeric values in the appropriate numeric field up to the *min* or *max* specified.